### PR TITLE
Export import fixes

### PR DESCRIPTION
**Pull request by AlexFreik**

### CODE REVIEW

File: frontend/components/blocks/UiNode.tsx [hunk 1]
```diff
@@ -36,16 +37,7 @@ export function TextBlock({ data }: { data: { node: TextNode } }) {
...
-		if (filepath == undefined) return;
-		node.filename = filepath.name;
-		const reader = new FileReader();
-		reader.onload = (event) => {
-			node.content = reader.result === null ? '' : reader.result.toString();
-		};
-		reader.onerror = () => {
-			console.log('Error loading');
-		};
-		reader.readAsText(filepath);
```
Comment (AlexFreik) [inline frontend/components/blocks/UiNode.tsx:48]: This can be separated into FileUtils.

File: frontend/components/document/Navbar.tsx [hunk 2]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
 	const [removeTrigger, setRemoveTrigger] = useState<boolean>(false);
 	const fileMenuStyle = styles.fileMenu + (isFileMenuOpen ? '' : ' ' + styles.fileMenuHidden);
 
-	const exportToFile = (extension: string) => {
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:26]: name can be misleading (taken you don't even have return type)

File: frontend/components/document/Navbar.tsx [hunk 3]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-
-		const graphJson = JSON.stringify(wiu.graph, null, 2);
-		const blob = new Blob([graphJson], { type: 'application/json' });
-		const href = URL.createObjectURL(blob);
-		const link = document.createElement('a');
-		link.href = href;
-		link.download = wiu.graph.name + extension;
-		document.body.appendChild(link);
-		link.click();
-		document.body.removeChild(link);
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:37]: this can be separated into `FileUtils`

File: frontend/components/document/Navbar.tsx [hunk 4]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-	const handleOpenFile = (fileToLoad: File | null) => {
-		if (fileToLoad) {
-			let jsonFile = new FileReader();
-			jsonFile.readAsText(fileToLoad);
-			jsonFile.onload = (evt: ProgressEvent<FileReader>) => {
-				if (typeof evt.target?.result === 'string') {
-					wiu.loadGraphToUi(JsonUtils.jsonStringToGraph(evt.target.result));
-				}
-			};
-		}
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:49]: It is usual mistake -- to have an if inside function that wraps the whole thing. 
It is better to either 
1. not to pass null here
2. have a line `if (fileToLoad == null) return;`

File: frontend/components/document/Navbar.tsx [hunk 5]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-
-	const handleOpenFile = (fileToLoad: File | null) => {
-		if (fileToLoad) {
-			let jsonFile = new FileReader();
-			jsonFile.readAsText(fileToLoad);
-			jsonFile.onload = (evt: ProgressEvent<FileReader>) => {
-				if (typeof evt.target?.result === 'string') {
-					wiu.loadGraphToUi(JsonUtils.jsonStringToGraph(evt.target.result));
-				}
-			};
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:48]: this can be separated into `FileUtils`

File: frontend/components/document/Navbar.tsx [hunk 6]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-		if (strFromStorage) {
-			JSON.parse(strFromStorage).map((item: AtlasGraph) =>
-				atlasStorage.push(JsonUtils.jsonToGraph(item)),
-			);
-		}
-		setRecentFromLocalStorage(atlasStorage.reverse());
+	const updateRecentElementsUi = () => {
+		setRecentFromLocalStorage(StorageUtils.getRecentGraphsFromLocalStorage().reverse());
 		setIsFileMenuOpen(false);
 		refSchemaName.current!.value = wiu.graph.name;
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:54]: it is better to give this function a name to make it more readable

File: frontend/components/document/Navbar.tsx [hunk 7]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-	}, [removeTrigger, wiu.graph.name]);
+	};
+	useEffect(updateRecentElementsUi, [removeTrigger]);
 
-	function getRecentGraphs(graphFromRecentList: AtlasGraph): JSX.Element {
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:78]: again. You don't return Graph**S**, but only one graph element

File: frontend/components/document/Navbar.tsx [hunk 8]
```diff
@@ -23,66 +24,46 @@ export default function Navbar({ wiu }: Props) {
...
-		if (graphFromRecentList.name === wiu.graph.name)
+	function getRecentGraphElement(graphFromRecentList: AtlasGraph): JSX.Element {
+		if (graphFromRecentList.name === wiu.graph.name) {
 			return <div key={graphFromRecentList.name}></div>;
+		}
 		return (
 			<div
 				key={graphFromRecentList.name}
 				className={styles.elementFileMenu}
-				onClick={() => wiu.loadGraphToUi(graphFromRecentList)}
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:85]: load have many meanings. it is better to use more precise 'replace`

File: frontend/components/document/Navbar.tsx [hunk 9]
```diff
@@ -136,37 +115,41 @@ export default function Navbar({ wiu }: Props) {
...
 				<div className={styles.icon}>
-					<Image src={settingsImg} layout={'responsive'} objectFit={'contain'} />
+					<Image src={settingsImg} objectFit={'contain'} />
 				</div>
 				<input
 					type={'file'}
 					ref={refOpenFile}
 					style={{ display: 'none' }}
-					onChange={(evt: ChangeEvent<HTMLInputElement>) =>
-						handleOpenFile(evt.target.files ? evt.target.files[0] : null)
```
Comment (AlexFreik) [inline frontend/components/document/Navbar.tsx:168]: again) ternary operator where `if` should be.
Also it is not safe to do implicit conversion of `evt.target.files` to boolean. It is better to be more precise `evt.target.files === null`.

File: frontend/utils/ServerUtils.ts [hunk 10]
```diff
@@ -54,44 +54,17 @@ abstract class ServerUtils {
...
-			updatedNodes.push(Object.assign(newNode, node));
-		}
-		return updatedNodes;
-	}
-
-	public static extractEdges(edges: object[]): AtlasEdge[] {
-		const updatedEdges: AtlasEdge[] = [];
-		edges.forEach((edge: object) => updatedEdges.push(Object.assign(AtlasEdge.build(), edge)));
-		return updatedEdges;
+		return JsonUtils.jsonToGraph(responseJson.graph);
```
Comment (AlexFreik) [inline frontend/utils/ServerUtils.ts:68]: they should be inside JsonUtils, not ServerUtils

File: frontend/utils/StorageUtils.ts [hunk 11]
```diff
@@ -1,20 +1,37 @@
 import AtlasGraph from './AtlasGraph';
+import JsonUtils from './JsonUtils';
 
 export default class StorageUtils {
+	private static readonly recentGraphStorageName = 'AtlasStorage';
+
 	public static saveGraphToLocalStorage(saveGraph: AtlasGraph) {
-		const strFromStorage: string | null = localStorage.getItem('AtlasStorage');
```
Comment (AlexFreik) [inline frontend/utils/StorageUtils.ts:5]: all magic values like `'AtlasStorage'` should be inside a variable

**Review by roman-timo — APPROVED** [review]
nice code!

**Pull request merged**